recipe/patch_yaml/nccl.yaml

```
@@ -18,3 +18,12 @@ then:
   - rename_depends:
       old: cudatoolkit
       new: __cuda
+---
+if:
+  name: nccl
+  timestamp_lt: 1758819475000
+  version: "2.26.5.1"
+then:
+  - replace_depends:
+      old: cuda-version >=12,<13.0a0
+      new: cuda-version >=12,<12.2
```

Admittedly there are not alpha versions, but just to be safe

```suggestion
      new: cuda-version >=12,<12.2a0
```